sirepo/template/srw.py

```diff
@@ -367,6 +370,36 @@ def _extract_coherent_modes(model, out_info):
     return out_file
 
 
+def _get_recent(data_filename, backup_data_filename):
+    if os.path.getmtime(backup_data_filename) > os.path.getmtime(data_filename):
+        return PKDict(
+            recent=backup_data_filename,
+            old=data_filename,
+        )
+    return PKDict(
+        recent=data_filename,
+        old=backup_data_filename,
+    )
+
+
+def _count_lines(text):
+    return len(pkio.read_text(text).splitlines())
+
+
+def _recent_has_sufficient_lines(recent, old):
+    return _count_lines(recent) >= _count_lines(old)
+
+
+def _check_backup(animation_meta_data):
+    b = animation_meta_data.filename + ".bkp"
```

Comment:
To simplify, have another nested function that checks existence and lines (both files may not exist) so the code looks like:

```py
def _best_data_file(primary):
    def _size(path):
        return path.size if path.exists() else 0

    p = pykern.pkio.py_path(primary)
    s = p.new(ext='dat.bkp')
    if _size(s) > _size(primary):
        p = s
    return p.basename
```